src/com/cft/contactmerge/io/XmlImporter.java

```diff
@@ -1,31 +1,132 @@
+import org.w3c.dom.NamedNodeMap;
+import org.xml.sax.SAXException;
 
 public class XmlImporter implements IImporter, Iterable<Contact> {
 
-    public void Load(String filename)
+    private Map<String, String> columnMap;
+    private NodeList dataNodes;
+
+    public void Load(String filename) throws FileNotFoundException
     {
-        // TODO: Add code
+        if (filename == null || filename.trim() == "")
+        {
+            throw new IllegalArgumentException("filename is required");
+        }
+
+        File inputFile = new File(filename);
+
+        if (!inputFile.exists() || inputFile.isDirectory())
+        {
+            throw new FileNotFoundException(String.format("Unable to find %s", filename));
+        }
+
+        InputStream inputStream = new FileInputStream(inputFile);
+
+        Load(inputStream);
+    }
+
+    public void Load(InputStream inputStream)
+    {
+        if (inputStream == null)
+        {
+            throw new IllegalArgumentException("inputStream is required");
+        }
+
+        try {
+            if (!(inputStream.available() > 0))
+                throw new UnknownFormatConversionException("inputStream not available");
+        }
+        catch (IOException e)
+        {
+            throw new UnknownFormatConversionException("inputStream not available");
+        }
+
+        try {
+            DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
+            DocumentBuilder dBuilder = dbFactory.newDocumentBuilder();
+            Document doc = dBuilder.parse(inputStream);
+
+            columnMap = getXmlAttribuateMap(doc.getElementsByTagName("Field"));
+            dataNodes = doc.getElementsByTagName("row");
+        }
+        catch(SAXException | IOException | ParserConfigurationException e)
+        {
+            throw new UnknownFormatConversionException(e.getMessage());
+        }
+    }
+
+    // TODO: Need to figure out a flexible way to map XML values with
+    // properties. This is a temporary solution so that we can start playing
+    // with data from GiftWorks.
+    private Map<String, String> getXmlAttribuateMap(NodeList nodes)
+    {
+        Map<String, String> map = new HashMap<String, String>();
+
+        for (int i=0; i < nodes.getLength(); i++)
+        {
+            NamedNodeMap mapItem = nodes.item(i).getAttributes();
+            map.put(mapItem.item(1).getNodeValue(), mapItem.item(0).getNodeValue());
+        }
+
+        return map;
+    }
+
+    // TODO: Need to figure out a flexible way to map XML values with
+    // properties. This is a temporary solution so that we can start playing
+    // with data from GiftWorks.
+    private Map<String, String> getContactValues(Node node)
+    {
+        Map<String, String> values = new HashMap<String, String>();
+
+        for (int j=0; j < node.getAttributes().getLength(); j++) {
+            values.put(node.getAttributes().item(j).getNodeName(), node.getAttributes().item(j).getNodeValue());
+        }
```

Comment:
I think it's better to use a for-each over a for loop if you aren't going to have to specifically interrupt the iteration flow of ++...

untested....
```
for (Node item: node.getAttributes()){
    values.put(item.getNodeName(), item.getNodeValue());
}
```

Reply:
That is what I tried originally but getAttributes() doesn't have an iterator. It returns NamedNodeMap not a Node.

Reply:
for (NamedNodeMap attributes: node.getAttributes()){
    values.put(attributes.getNodeName(), attributes.getNodeValue());
}

maybe?